test/hotspot/gtest/runtime/test_os.cpp

```diff
@@ -587,3 +587,85 @@ TEST_VM(os, find_mapping_3) {
   }
 }
 #endif // _WIN32
+
+TEST_VM(os, pagesizes) {
+  ASSERT_EQ(os::min_page_size(), 4 * K);
+  ASSERT_LE(os::min_page_size(), (size_t)os::vm_page_size());
+  // The vm_page_size should be the smallest in the set of allowed page sizes
+  // (contract says "default" page size but a lot of code actually assumes
+  //  this to be the smallest page size; notable, deliberate exception is
+  //  AIX which can have smaller page sizes but those are not part of the
+  //  page_sizes() set).
+  ASSERT_EQ(os::page_sizes().smallest(), (size_t)os::vm_page_size());
+  // The large page size, if it exists, shall be part of the set
+  if (UseLargePages) {
+    ASSERT_GT(os::large_page_size(), (size_t)os::vm_page_size());
+    ASSERT_TRUE(os::page_sizes().contains(os::large_page_size()));
+  }
+  os::page_sizes().print_on(tty);
+  tty->cr();
+}
+
+static const int min_page_size_log2 = exact_log2(os::min_page_size());
+static const int max_page_size_log2 = (int)(sizeof(size_t) * 8);
+
+TEST_VM(os, pagesizeset_test_range) {
+  for (int bit = min_page_size_log2; bit < max_page_size_log2; bit ++) {
+    for (int bit2 = min_page_size_log2; bit2 < max_page_size_log2; bit2 ++) {
+      const size_t s =  (size_t)1 << bit;
+      const size_t s2 = (size_t)1 << bit2;
+      //tty->print_cr(SIZE_FORMAT " - " SIZE_FORMAT, s, s2);
+      os::PageSizes pss;
+      // Empty set
+      for (int bit3 = min_page_size_log2; bit3 < max_page_size_log2; bit3 ++) {
+        const size_t s3 = (size_t)1 << bit3;
+        ASSERT_FALSE(pss.contains(s3));
+      }
+      ASSERT_EQ((size_t)0, pss.smallest());
+      ASSERT_EQ((size_t)0, pss.largest());
```

Comment:
This doesn't have to be inside a double nested for loop. This could be single check done once.